lib/active_merchant/billing/gateways/card_stream.rb

```diff
@@ -283,6 +284,20 @@ def add_threeds_required(post, options)
         add_pair(post, :threeDSRequired, options[:threeds_required] || @threeds_required ? 'Y' : 'N')
       end
 
+      def add_global_threeds2_mpi_data(post, options)
+        return unless three_d_secure = options[:three_d_secure]
+
+        add_pair(post, :threeDSEnrolled, three_d_secure[:enrolled] == 'true' ? 'Y' : 'N')
```

Comment:
💬 Comment:

add_pair is going to work any way, but I don't see the need of it here, you don't have any reason to thing that once you detected 3ds hash that the other params could be missing.